common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -341,6 +340,15 @@ public boolean contains(final UTF8String substring) {
     return false;
   }
 
+  public boolean contains(final UTF8String substring, int collationId) {
+    if (collationId == CollationFactory.LOWERCASE_COLLATION_ID) {
```

Comment:
What if we pass DEFAULT_COLLATION_ID here?

Reply:
It's fine to have a short circuit and have `stringExpression` directly call `containsBinary`. I just think that  `contains` should work for all collations, including `UCS_BASIC`.

Reply:
Ahh, I see. Makes sense, I do agree that `contains(final UTF8String substring, int collationId)` should be able to handle all collations, even if it's not currently invoked for DEFAULT_COLLATION_ID. I added this in the latest commit!

However, that same default option that works for DEFAULT_COLLATION_ID (UCS_BASIC) also works for other binary collations supported by CollationFactory (currently: UNICODE).

I suppose we should go ahead and cover them all here using:
`if (CollationFactory.fetchCollation(collationId).isBinaryCollation)`
instead of just
` if (collationId == CollationFactory.DEFAULT_COLLATION_ID)`